include/solvers/time_solver.h

```diff
@@ -225,6 +225,19 @@ class TimeSolver : public ReferenceCountedObject<TimeSolver>
    */
   void set_solution_history(const SolutionHistory & _solution_history);
 
+  /**
+   * A setter function for the replace_stored_entry boolen owned by
+   * the solution history object
+   * */
+  void set_solution_history_replace_stored_entry(bool val)
+  { solution_history->set_replace_last_stored(val); }
+
+  /**
+   * A getter function that returns a reference to the solution history
+   * object owned by TimeSolver
+   * */
+  std::shared_ptr<SolutionHistory> get_solution_history();
```

Comment:
So far we only need this for the outer time solver in the adaptive time stepping classes, right?  Would it make sense to make it protected at first, so we have a little more flexibility if we change our minds about the API in the future?